2-promises/solution.js

```diff
@@ -24,16 +24,37 @@ const id = yourRandomMethod() //third run
 7. log the resultant fullname, or the error, at the end
 */
 
+const firstnames = require("./firstnames.js");
+const lastnames = require("./lastnames.js");
+
 function solution() {
     // YOUR SOLUTION GOES HERE
+    let first = "";
+    let last = "";
 
     // You generate your id value here
+    const rng = Math.round(Math.random() * 100) * (Math.round(Math.random()) === 1 ? 1 : -1) ;// randomly make it negative
 
     // You call the lastnames method with your id
+    lastnames(rng)
 
     // You call the firstname method
-
+    .then((lastname)=>{
+        last = lastname;
+        return firstnames(lastname);
+    })
+    .then(firstname=>{
+        first = firstname;
+    })
```

Comment:
I think we don't need to use a variable for last and first name here, you can rely just in the nested promises.

I recommend you to use the method then with first name method